osf/utils/migrations.py

```diff
@@ -390,6 +390,39 @@ def create_schema_blocks_for_question(state, rs, question, sub=False):
         split_options_into_blocks(state, rs, question, schema_block_group_key)
 
 
+
+def create_schema_blocks_for_simple_schema(schema):
+    from osf.models import RegistrationSchemaBlock
+    schema_block_group_key = ''
+    page_num = 0
+    q_num = 1
+    for block in schema.schema['blocks']:
+        block_type = block['block_type']
+
+        if block_type == 'question-label':
+            schema_block_group_key = generate_object_id()
+        if block_type == 'section-heading':
+            schema_block_group_key = generate_object_id()
+        if block_type == 'subsection-heading':
+            schema_block_group_key = generate_object_id()
+        if block_type == 'paragraph':
+            schema_block_group_key = ''
+        if block_type == 'page-heading':
+            page_num += 1
+            q_num = 1
+            schema_block_group_key = ''
+
+        schemablock, created = RegistrationSchemaBlock.objects.update_or_create(
+            schema_id=schema.id,
+            **block
+        )
+        if created:
+            schemablock.registration_response_key = block['registration_response_key']
+            schemablock.schema_block_group_key = schema_block_group_key
+
+        schemablock.save()
+        q_num += 1
```

Comment:
Is this even being used anymore? Looks like it has been replaced by explicitly coding the response_key in the json blocks. Wherever the key is being generated, this should only be incremented in the case of an input field, otherwise you're creating confusing gaps in numbering.